designs/ClientDevProcess.md

```diff
@@ -0,0 +1,145 @@
+# NuGet Client Team Dev Process
+
+- Status: **Reviewing**
+- Author: [Kat Marchan](https://github.com/zkat)
+
+# Issue
+
+[#8994](https://github.com/NuGet/Home/issues/8994)
+
+# Background
+
+For a while now, the NuGet Client team has been using [ZenHub](https://zenhub.com) as part of its development process. At the same time, while ZenHub provides a lot of features, the team isn't taking advantage of some of the nicer ones, such as its charting capabilities and its Roadmap feature. It also has other features, like Blockers and Epics, which the team does use but could be using more often, or differently.
+
+This proposal is aimed at overhauling the Client Team's usage of ZenHub and further streamline its public development process to increase velocity, as well as increase visibility into what's being done, and what needs to be done. It also describes existing processes, so this document may be used as a reference for development flow.
+
+# Proposal
+
+## OKR Tracking
+
+- Link: https://github.com/NuGet/Home#workspaces/nuget-client-team-55aec9a240305cf007585881/roadmap?repos=29996513
+
+One of the main additions in this proposal is that it adds explicit tracking of OKRs into our ZenHub process. The way this works is by using ZenHub's Roadmap feature.
+
+Each quarter, a new Project should be created for each quarterly Objective, and labeled as such. Inside each Project should be any Epics created for those particular KRs, with dates adjusted appropriately. Additionally, a "Non-KR Work" Project should be created, with any non-KR epics, such as bugfixes and such. Ideally, all work will exist under some quarterly epic and be displayed on the roadmap.
+
+## Epic Tracking
+
+The Client Team uses Epics to track high-level status of various issues. Epics should be created for tracking large verticals of work, or when there's many disparate subtasks that will be involved and potentially assigned to multiple people, or involve multiple PRs to complete. As described in [OKR Tracking](#okr-tracking), they should also be created to track specific KRs.
+
+Another way to think about Epics is: if there is a customer-facing feature, an Epic should be the parent of all tasks from spec'ing to writing the actual code to test to insertion to shipping and writing docs.
+
+Concretely, these are GitHub issues tagged as Epic, and they navigate the board slightly different from regular issues:
+
+1. When any issue under an Epic is In Progress, the associated Epic should be moved to the In Progress pipeline as well.
+2. Epics should remain under "In Review" or "Validating" until the related change is available to customers. That is, they shouldn't be closed until released.

```

Comment:
May be an additional tag on Epic that differentiates from internal or random big chunk of work from customer facing features that we would to track on the `Roadmap` project?

Reply:
What tag would you like? And would it be ok to let PM be in charge of adding it? :)

Reply:
I was thinking about having a label like `Feature` or anything else that suits this design. IMO, such an Epic should be assigned to PMs. And hence once we decide on the label, anybody including PMs can add them.

Reply:
I feel like feature is an overloaded term :) 
There are lots of things that are features and should be tracked as such, not just ones that need to part of the roadmap. 

I'd suggest something explicit...call it roadmap if the only purpose on the label is to indicate that the epic in question is part of the roadmap.

Reply:
What's an example of a feature that wouldn't be on the public roadmap?

Reply:
If we look at https://github.com/NuGet/home/issues?q=is%3Aopen+is%3Aissue+label%3AType%3AFeature for example. 

There are 200+ issues labeled as Type:Feature. Now if we review those, probably not all of those are truly features.

Some of these are engineering features that affect a few people, but are still from a product perspective features. They are a significant, additive part of the product.
https://github.com/NuGet/Home/issues/8719 In this case, the product is our library. 
https://github.com/NuGet/Home/issues/7917 Product is msbuild, few people would use it, but useful. 
https://github.com/NuGet/Home/issues/7709

I feel like adding all these in the public roadmap, might introduce visual overload. 
Finally, there are certain bug fixes that deserve to be part of the public roadmap as well :) The Task Cancelled work is one that I can remember.